app/src/main/res/layout/event_create_form.xml

```diff
@@ -624,18 +624,17 @@
                         android:textSize="@dimen/text_size_normal" />
 
                     <CheckBox
-                        android:id="@+id/paypal_payment"
                         android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
-                        android:onCheckedChanged="@{ (switch, checked) -> event.setCanPayByPaypal(checked) }"
+                        android:checked="@={ event.canPayByPaypal }"
                         android:text="@string/accept_payment_through_paypal" />
 
                     <android.support.design.widget.TextInputLayout
                         android:id="@+id/paypal_email_layout"
                         android:layout_width="match_parent"
                         android:layout_height="wrap_content"
                         android:padding="@dimen/spacing_extra_small"
-                        android:visibility="@{ paypalPayment.checked ? View.VISIBLE : View.GONE }">
+                        android:visibility="@{ event.canPayByPaypal ? View.VISIBLE : View.GONE }">
 
```

Comment:
@iamareebjamal @Masquerade0097 I am trying to get this TextInputLayout to show/hide but it reacts correctly only at first and doesn't change after I further check checkbox. I have used `@=` in checkbox so the value `event.canPayByPaypal` is also updating.
Any idea why?

Reply:
@mishuvs I think the problem is similar to this https://stackoverflow.com/questions/20040542/android-development-checkbox-setchecked-not-working.
Use `android:onCheckChanged`, it will work as expected.